internal/command/init.go

```diff
@@ -55,12 +56,22 @@ func (c *InitCommand) Run(args []string) int {
 		return 1
 	}
 
+	path, err := ModulePath(initArgs.Args)
+	if err != nil {
+		diags = diags.Append(err)
+		view.Diagnostics(diags)
+		return 1
+	}
+	rootMod, _ := c.loadSingleModuleWithTests(path, initArgs.TestsDirectory)
+	// We purposefully ignore any diagnostics returned here. They will be encountered downstream,
+	// when the 'run' logic below is executed. If we return early due to error diagnostics here we
+	// will break the order that errors are expected to be raised in.
```

Comment:
There's a problem with this approach- loading config isn't idempotent.

When you parse HCL files using `(p *Parser) ParseHCL` from hashicorp/hcl it stores the parsed file in an internal map. Any diagnostics raised from parsing the file the first time are only returned when the file is first parsed. If you re-parse the same file [the old parsed data is returned with no associated diagnostics](https://github.com/hashicorp/hcl/blob/0268c1604970208300c320f4129fccd13a9e2b04/hclparse/parser.go#L55-L58).

Reply:
This problem was apparent through `TestInit_invalidSyntaxBackendAttribute` failing - the diagnostics we expect in the test get lost between the 1st and 2nd time the config is parsed.